crystal: update to 0.29.0.

gcat432 opened the pull request.

Hoshpak commented:
Continuing the discussion from #12189:

You don't need to cross-compile to test i686 from an x86_64 host. Creating a separate i686 masterdir via binary-bootstrap and using that one is enough. I would like to have compilation tested before merging because i686 is not a priority for many upstream projects and things tend to break there very often nowadays.

gcat432 commented:
Thank you for the tip!

Everything seems to went fine on i686:
```
...
=> Creating crystal-0.29.0_1.i686.xbps for repository /host/binpkgs/crystal-0.29.0 ...
=> Creating crystal-32bit-0.29.0_1.x86_64.xbps for repository /host/binpkgs/crystal-0.29.0/multilib ...
=> crystal-0.29.0_1: running post-pkg hook: 00-register-pkg ...
=> Registering new packages to /host/binpkgs/crystal-0.29.0
index: added `crystal-0.29.0_1' (i686).
index: 1 packages registered.
=> Registering new packages to /host/binpkgs/crystal-0.29.0/multilib (x86_64)
index: added `crystal-32bit-0.29.0_1' (x86_64).
index: 1 packages registered.
```
